extensions.html

```diff
@@ -445,16 +507,122 @@ <h2>
         </dt>
         <dd>
           A list of touch events generated from all touch surfaces.
-          <code>null</code> if the device does not support touch events.
+          If the device does not support touch events, MUST be set to `null`.
         </dd>
       </dl>
+      <section>
+        <h3>
+          Receiving inputs
+        </h3>
+        <p>
+          This section supplements the <a href="https://w3c.github.io/gamepad/#receiving-inputs">
+          Receiving inputs</a> section of the main 
+          <a href="https://w3c.github.io/gamepad/">Gamepad specification</a>.
+        </p>
+        <p>
+          In addtion to the steps defined in the main 
+          <a href="https://w3c.github.io/gamepad/">Gamepad specification</a>.
+          When the system receives new touch input values,
+          <dfn data-lt="updating touchEvents">update touchEvents</dfn>
+          by running the following steps:
+        </p>
+        <ol>
+          <li>Let |surfaceDimensionX:unsigned long| be an {{unsigned long}} value
+            representing the maximum x dimention of the touch surface on the |gamepad|.
+          </li>
+          <li>Let |surfaceDimensionY:unsigned long| be an {{unsigned long}} value
+            representing the maximum y dimention of the touch surface on the |gamepad|.
+          </li>
+          <li>Let |touchInputs:list| be a [=list=] of objects representing
+            the most recent logical touch input values reported by the |gamepad|.
+          </li>
+          <li>[=list/For each=] |touchData| in |touchInputs:list|:
+            <ol>
+              <li>
+                <p>Let |normalizedX:float| be a value calculated from the current
+                  touch input x coordinate in relationship to the surfaceDimensionX
+                  and normalized to the range [-1.0, 1.0] where -1.0 is the leftmost
+                  coordinate and 1.0 is the rightmost coordinate.</p>
+                <p>
+                <p>Possible implementation:</p>
+                <ul>
+                  <li>`normalizedX = (2.0 * touchData.x / surfaceDimensionX) - 1`</li>
+                </ul>
+                </p>
+              </li>
+              <li>
+                <p>Let |normalizedY:float| be a value calculated from the current
+                  touch input y coordinate in relationship to the surfaceDimensionY
+                  and normalized to the range [-1.0, 1.0] where -1.0 is the topmost
+                  coordinate and 1.0 is the bottommost coordinate.
+                </p>
+                <p>
+                <p>Possible implementation:</p>
+                <ul>
+                  <li>`normalizedY = (2.0 * touchData.y / surfaceDimensionY) - 1`</li>
+                </ul>
+                </p>
+              </li>
+              <li>Let |touchEvent:GamepadTouch| be a {{GamepadTouch}}.
+              </li>
+              <li>Set |touchEvent|.{{GamepadTouch/touchId}} to be
+              `touchData.touchID`.
```

Comment:
@nondebug I manually added the link, it appears that the dfn is not exported (or possibly, probably, I am not using xref correctly). I am also a little concerned that we are making the spec too restrictive to implementers, e.g. I am working on the DS4 GamepadTouch implementation in chromium and find that normalizing the touchId is much easier at the ds4_controller than the device/gamepad (this may be a bad assumption on my part that when you refer to an "internal slot" it would translate to that object). Requiring the nextTouchId in device/gamepad might cause difficulty in implementing other touch controllers. The specification just needs to be clear that implementers support 32 bit value starting at 0 and incrementing with each new touch event not associated with an existing active touch point. How this is accomplished should be open.
I will try and add some "Usage Examples" to better explain how applications will interact with the API.